src/modules/gnd_pos_control/GroundRoverPositionControl.cpp

```diff
@@ -331,13 +319,17 @@ GroundRoverPositionControl::task_main()
 	}
 
 	/* wakeup source(s) */
-	px4_pollfd_struct_t fds[2];
+	px4_pollfd_struct_t fds[4];
 
 	/* Setup of loop */
 	fds[0].fd = _params_sub;
 	fds[0].events = POLLIN;
 	fds[1].fd = _global_pos_sub;
 	fds[1].events = POLLIN;
+	fds[2].fd = _manual_control_sub;
```

Comment:
This likely isn't what you want here. These topics publish asynchronously, so the main loop in this module is going to run much faster than intended.

Limiting it to `_vehicle_attitude_sub` alone is probably sufficient as long as you also check the global position validity before using any of that data (when required). Ideally there would also be a mechanism to run real manual without depending on attitude publications, but that's not going to matter in practice.

Reply:
I could implement the real manual mode fairly easily. The reason I did it this way is because the original version of the code published `actuator_controls_0` at a constant 250Hz, and I tried to keep the functionality as similar as possible. Would it be an issue to publish this topic asynchronously?

Reply:
Probably easier to talk about it on a call.

Reply:
I have made the changes we talked about on the call. `actuator_controls_0` now publishes at a constant 250Hz, regardless of incoming messages.